inductiva/api/methods.py

```diff
@@ -138,29 +138,34 @@ def upload_input(api_instance: TasksApi, task_id, original_params,
         original_params: Params of the request passed by the user.
         type_annotations: Annotations of the params' types.
     """
-
-    input_zip_path, zip_file_size = prepare_input(task_id, original_params,
-                                                  type_annotations)
-
-    remote_input_zip_path = f"{storage_path_prefix}/{task_id}/input.zip"
-    url = storage.get_signed_urls(
-        paths=[remote_input_zip_path],
-        operation="upload",
-    )[0]
-
-    with tqdm.tqdm(total=zip_file_size,
-                   unit="B",
-                   unit_scale=True,
-                   unit_divisor=1000) as progress_bar:
-        upload_file(api_instance, input_zip_path, "PUT", url, progress_bar)
-        notify_upload_complete(
-            api_instance.notify_input_uploaded,
-            path_params={"task_id": task_id},
-        )
-
-    logging.info("Local input directory successfully uploaded.")
-    logging.info("")
-    os.remove(input_zip_path)
+    # Use the blocking task context
+    with blocking_task_context(api_instance, task_id):
```

Comment:
Instead of calling the blocking_task_context inside of the function, i think it would look better if you call the function with the context wrapped around it. 

Like

```python
with blocking_task_content(...):
    upload_input(....)
```

instead of 

```python
def upload_input(...):
    with blocking_task_context(...):
```

Doing this you don't need to change the function definition